imap_processing/ccsds/excel_to_xtce.py

```diff
@@ -0,0 +1,426 @@
+"""
+Convert an Excel file of packet definitions into the XTCE format.
+
+This script reads in an Excel file containing packet definitions and converts
+them into an XTCE file.
+
+.. code::
+  imap_xtce /path/to/excel_file.xlsx --output /path/to/output.xml
+"""
+
+import argparse
+import xml.etree.ElementTree as Et
+from importlib.util import find_spec
+from pathlib import Path
+
+import pandas as pd
+
+_CCSDS_PARAMETERS = [
+    {
+        "name": "VERSION",
+        "lengthInBits": 3,
+        "description": "CCSDS Packet Version Number (always 0)",
+    },
+    {
+        "name": "TYPE",
+        "lengthInBits": 1,
+        "description": "CCSDS Packet Type Indicator (0=telemetry)",
+    },
+    {
+        "name": "SEC_HDR_FLG",
+        "lengthInBits": 1,
+        "description": "CCSDS Packet Secondary Header Flag (always 1)",
+    },
+    {
+        "name": "PKT_APID",
+        "lengthInBits": 11,
+        "description": "CCSDS Packet Application Process ID",
+    },
+    {
+        "name": "SEQ_FLGS",
+        "lengthInBits": 2,
+        "description": "CCSDS Packet Grouping Flags (3=not part of group)",
+    },
+    {
+        "name": "SRC_SEQ_CTR",
+        "lengthInBits": 14,
+        "description": "CCSDS Packet Sequence Count "
+        "(increments with each new packet)",
+    },
+    {
+        "name": "PKT_LEN",
+        "lengthInBits": 16,
+        "description": "CCSDS Packet Length "
+        "(number of bytes after Packet length minus 1)",
+    },
+]
+
+
+class XTCEGenerator:
```

Comment:
Is this a complete re-write of telemetry_generator? Or is this a rename/move?

Reply:
It is a pretty significant refactor, but then I did do a `git mv ...` and it still says it is an added/deleted file for some reason rather than keeping the "diff" from before. I think this might be because it was such a large refactor and so many lines changed that it was easier for git to call it new/deleted.

Reply:
Ok, just wanted to know what was being updated.